xwiki-commons-core/xwiki-commons-configuration/xwiki-commons-configuration-default/src/main/java/org/xwiki/configuration/internal/SystemPropertiesConfigurationSource.java

```diff
@@ -0,0 +1,165 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.configuration.internal;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import org.xwiki.component.annotation.Component;
+import org.xwiki.configuration.ConfigurationSaveException;
+
+/**
+ * {@link System} properties based configuration source.
```

Comment:
Wrong `@link`?

Reply:
Same as above, maybe would be useful to have an example in the javadoc to define a variable?

Reply:
It's not easy to give an accurate example at this level as it's not supposed to be used directly, it's more a tool used by other configuration sources (and the prefix they choose) to overwrite their properties.

Reply:
> Wrong @link?

Why ? That's where the properties are coming from.

Reply:
My bad, since it's a java lang and I wasn't seeing a FQN I thought it was a typo.